pkg/identity/deploy.go

```diff
@@ -649,26 +650,156 @@ func roleBindingResource(roleBindingName, roleName, serviceAccountName, serviceA
+			return errors.Wrap(err, "failed to get existing secret")
 		}
 
-		_, err = clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(ctx, &crd, metav1.CreateOptions{})
+		_, err = clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
 		if err != nil {
-			return errors.Wrapf(err, "failed to create crd %s", crd.Spec.Names.Plural)
+			return errors.Wrap(err, "failed to create secret")
 		}
 
 		return nil
 	}
 
-	// no patch necessary
+	// no patch needed
+
+	return nil
+}
+
+func postgresSecretResource(secretName string) *corev1.Secret {
+	generatedPassword := ksuid.New().String()
+
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   secretName,
+			Labels: kotsadmtypes.GetKotsadmLabels(AdditionalLabels),
+		},
+		Data: map[string][]byte{
+			"password": []byte(generatedPassword),
+		},
+	}
+}
+
+func ensurePostgresJob(ctx context.Context, clientset kubernetes.Interface, namespace string, registryOptions *kotsadmtypes.KotsadmOptions) error {
+	job := postgresJobResource(DexPostgresJobName, namespace, registryOptions)
+
+	_, err := clientset.BatchV1().Jobs(namespace).Get(ctx, DexPostgresJobName, metav1.GetOptions{})
+	if err != nil {
+		if !kuberneteserrors.IsNotFound(err) {
+			return errors.Wrap(err, "failed to get existing postgres job")
+		}
+
+		_, err = clientset.BatchV1().Jobs(namespace).Create(ctx, job, metav1.CreateOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to create job")
+		}
+
+		return nil
+	}
+
+	// TODO: recreate job?
 
 	return nil
 }
 
+func postgresJobResource(name string, namespace string, registryOptions *kotsadmtypes.KotsadmOptions) *batchv1.Job {
+	image := "postgres:10.7"
+	imagePullSecrets := []corev1.LocalObjectReference{}
+	if registryOptions != nil {
+		if s := kotsadmversion.KotsadmPullSecret(namespace, *registryOptions); s != nil {
+			image = fmt.Sprintf("%s/postgres:%s", kotsadmversion.KotsadmRegistry(*registryOptions), kotsadmversion.KotsadmTag(*registryOptions))
+			imagePullSecrets = []corev1.LocalObjectReference{
+				{
+					Name: s.ObjectMeta.Name,
+				},
+			}
+		}
+	}
+
+	job := &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "batch/v1",
+			Kind:       "Job",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: types.GetKotsadmLabels(),
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   name,
+					Labels: types.GetKotsadmLabels(),
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy:    "OnFailure",
+					ImagePullSecrets: imagePullSecrets,
+					Containers: []corev1.Container{
+						{
+							Image:           image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Name:            name,
+							Command: []string{
+								"psql",
+							},
+							Args: []string{
+								"-h",
```

Comment:
have you tested this in prod? are you sure the kotsadm user has perms to create dbs in prod?

Reply:
i have tested this with a `kots install` in my codeserver and it was working just fine. kotsadm user is a superuser